### DISCUSSION TITLE
ColumnSpacing and RowSpacing for UniformItemsLayout

### DISCUSSION THREAD

**Discussion by MauiUIui** [CommunityToolkit/Maui#1596] [New Feature Discussions]
First of all I'd like to say that I already had this subject in another [General] discussion and it was closed due to the lack of votes. Being too deep in the matter, the people of CommunityToolKit seem to have the feeling that most of the .NET developers actually use MAUI and that most of those actually use the CommunityToolKit and that most of those actually go to github and have the time to read the discussions and that most of those will certainly vote on those discussions. More on this down below.

### Now for the issue itself:
I really like the sadly rather less known UniformGrid of WPF and I'm very thankful you guys made the effort to make the functionality also available for MAUI. However doesn't it make sense to make it more MAUI-compatible by including a "Spacing" property. For reference WPF doesn't have spacing in any of its panels, including its most improtant one (Grid), but MAUI does have RowSpacing and ColumnSpacing built-in in the Grid and Spacing in other layouts.

This very much needed control doesn't get much attention because of the fact, that it lacks those modern features that the MAUI controls has, such as the spacing.

I hope this as a feature discussion would get the holy 7 votes that it didn't get as a [General] discussions in order for it to not be closed! 

### A note about the closing criteria you proposed [here](https://youtu.be/N9wMcBP4jtg?t=2889):

IMHO this out vetoing discussions due to the lack of voting numbers has the negative impact of losing great suggestions. More or less the reason of this is the fact that MAUI itself is still not widely used, let alone the MauiCommunityToolKit. The great majority of developers who currently actually use it and don't go to less buggy, well known solutions like say Flutter, are actually .NET fanatics or at least developers with great .NET experience, whose ideas and observations are mostly based on in depth practical "field" UI experience. This "abrupt" closing is causing us to simply lose this. Not even big companies with widely used products do this, and they a huge "fan base" which would more justify such filtering. We don't have this fan base to even think about something like this. Keeping up with this would cause people like me who make the effort and invest time to open a discussion to stop doing that.

**Comment by bijington** [comment]
@MauiUIui I can understand how it can be annoying that a feature idea, which is important to yourself, is closed can make you feel frustrated. Our intention as maintainers of this toolkit is never to intentionally cause frustration to others, especially not the users of the toolkit. We as a small team, with limited free time to spend, need to be pragmatic about what we can spend our time on, and as all organisations do, we have to plan to fit the biggest audience and try to provide the most of amount of value to our users. I can assure you that every big company I have worked for follows processes like this, otherwise they would end up with a never ending list of functionality that people not even want or use. We took the decision to close off discussions or proposals that don't have enough buy in from the community because at the end of the day this toolkit is built for the community. We also decided that we wanted to close discussions/issues down in order to prevent the authors from being misled into the impression that their requests will be fulfilled and also to aid our own sanity. I can honestly say how overwhelming it is when there is a constant stream of requests or issues coming in.

I know that we have this discussion on the topic: https://github.com/CommunityToolkit/Maui/discussions/1445 which I feel rather ironically that it should not be closed down and left open to make it easier to understand how we intend on discussions to work.

All that being said I do believe the proposal that you have put forward could well be useful to others, I'll be honest I haven't used the `UniformItemsLayout` in any of my app for it to benefit me but I suspect people that have used it would find it useful. I do also believe that we as team failed to engage with you in the discussion.

I feel this discussion started rather off topic so I am going to close this one down. I will reopen the original (https://github.com/CommunityToolkit/Maui/discussions/1152) and engage with you there. I will add though, if developers don't find the time to come to the repository looking for this functionality then it may well struggle to reach the holy 7 votes as you put it. But you will be starting with 3 votes already...
